private/bufpkg/bufapimodule/module_resolver.go

```diff
@@ -57,26 +57,12 @@ func (m *moduleResolver) GetModulePin(ctx context.Context, moduleReference bufmo
 		}
 		return nil, err
 	}
-	var targetModulePin bufmoduleref.ModulePin
-	moduleIdentity := getModuleIdentity(moduleReference)
-	for _, protoModulePin := range protoModulePins {
-		modulePin, err := bufmoduleref.NewModulePinForProto(protoModulePin)
-		if err != nil {
-			return nil, err
-		}
-		if getModuleIdentity(modulePin) == moduleIdentity {
-			if targetModulePin != nil {
-				return nil, fmt.Errorf("resolved multiple module pins for %q", moduleIdentity)
-			}
-			targetModulePin = modulePin
-		}
-	}
-	if targetModulePin == nil {
-		return nil, fmt.Errorf("resolved module pins did not contain module %q", moduleIdentity)
-	}
-	return targetModulePin, nil
-}
-
-func getModuleIdentity(moduleIdentity bufmoduleref.ModuleIdentity) string {
-	return moduleIdentity.Remote() + "/" + moduleIdentity.Owner() + "/" + moduleIdentity.Repository()
+	return bufmoduleref.NewModulePin(
+		moduleReference.Remote(),
+		moduleReference.Owner(),
+		moduleReference.Repository(),
+		bufmoduleref.Main,
```

Comment:
Main does not seem right here, can we make this empty? Or take branch from `repositoryCommit`?